File: node/day1.js
const { exec } = require('child_process')
const tools = require('./tools')

const expenses = tools.readFromFile('../inputs/day1.txt')
                  .map((v) => parseInt(v))

function getExpensesTwo(expenses) {
  for (let x = 0; x < expenses.length; x++) {
    for (let y = x; y < expenses.length; y++) {
      if (expenses[x] + expenses[y] === 2020) {
        console.log('found')
        return expenses[x] * expenses[y]
      }
    }
  }
}

function getExpensesThree(expenses) {
  for (let x = 0; x < expenses.length; x++) {
    for (let y = x; y < expenses.length; y++) {
      for (let z = y; z < expenses.length; z++) {
        if (expenses[x] + expenses[y] + expenses[z] === 2020) {
          return expenses[x] * expenses[y] * expenses[z]
        }
      }
    }
  }
}

// Playing around with .some and .slice
function testje(expenses) {
  let value = null
  expenses.some((x, v) => {
    if (value) return true
    expenses.slice(v).some((y) => {
      if (x + y === 2020) {
        value = x * y
        return true
      }
    })
  })
  return value
}

console.log(`Part 1: ${getExpensesTwo(expenses)}`)
console.log(`Part 2: ${getExpensesThree(expenses)}`)

console.log(`Test for part 1: ${testje(expenses)}`)

File: day1.py
#!/usr/bin/python3
from shared import read_from_file

expenses = list(map(lambda x: int(x), read_from_file('inputs/day1.txt')))

def getExpensesTwo(expenses):
  # return [x * y for x in expenses for y in expenses if x + y == 2020][0] # Not sure if pretty
  for x in expenses:
    for y in expenses:
      if x + y == 2020:
        return x * y

def getExpensesThree(expenses):
  for x in expenses:
    for y in expenses:
      for z in expenses:
        if x + y + z == 2020:
          return x * y * z

print("Part 1", getExpensesTwo(expenses))
print("Part 2", getExpensesThree(expenses))